datastore/elastic.go

```diff
@@ -0,0 +1,155 @@
+package datastore
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/bar41234/bar_book_service/models"
+	"github.com/olivere/elastic/v7"
+	"github.com/pkg/errors"
+	"strconv"
+	"strings"
+)
+
+const (
+	indexName = "bar_books_store_2"
+	querySize = 100
+)
+
+type elasticBookManager struct {
+	client *elastic.Client
+}
+
+func NewElasticBookManager(url string) (*elasticBookManager, error) {
+	client, err := elastic.NewClient(elastic.SetURL(url))
+	if err != nil {
+		return nil, errors.New("Elastic client creation was failed")
+	}
+	return &elasticBookManager{client: client}, nil
+}
+
+func (e *elasticBookManager) GetBook(id string) (*models.Book, error) {
+	ctx := context.Background()
+	res, err := e.client.Get().Index(indexName).Id(id).Do(ctx)
+	if err != nil {
+		return nil, err
+	}
+	var book models.Book
+	err = json.Unmarshal(res.Source, &book)
+	if err != nil {
+		return nil, err
+	}
+	book.Id = res.Id
+	return &book, nil
+}
+
+func (e *elasticBookManager) AddBook(book models.Book) (string, error) {
+	ctx := context.Background()
+	jsonBook, err := json.Marshal(book)
+	if err != nil {
+		return "", err
+	}
+	bookJason := string(jsonBook)
+	res, err := e.client.Index().
+		Index(indexName).
+		BodyJson(bookJason).
+		Do(ctx)
+	if err != nil {
+		return "", err
```

Comment:
It is more classic to return `*string` if there is a case where the function returns an error (like here) - this is not critical